http-core/src/main/scala/org/apache/pekko/http/impl/util/StreamUtils.scala

```diff
@@ -260,6 +261,15 @@ private[http] object StreamUtils {
       }
   }
 
+  /**
+   * Similar idea as [[FlowOps.statefulMap]] but simpler,
+   * which the state is captured instead of a passing in parameter, when
+   * the Flow completes, the state is lost, if you prefer additional emitting on complete,
+   * use [[FlowOps.statefulMap]] instead.
+   */
+  def statefulMap[T, U](functionConstructor: () => T => U): Flow[T, U, NotUsed] =
```

Comment:
@pjfanning @jrudolph @mdedetrich Do you think we can put this in pekko-core too? we just need another name.

Reply:
`lazyMap`?

Reply:
please raise an issue in pekko-core

I have little objections to new methods. I just get worried when existing methods are changed and if that could have negative consequences.

Reply:
I don' t have a better name too...